### ISSUE TITLE
Is it working?

### ISSUE THREAD

**Issue by bertdeelman** [DMailMan/MMM-MQTTfloorplan#2]
Hi,

I have a really simple configuration to start with. 

{
    topic: 'homie/homey/hal-sensor-magnet2/alarm-contact',
    label: 'hal-toiletdeur',
    type: 'door',
    display: { left: 300, top: 350 },
},
{
    topic: 'homie/homey/hal-sensor-magnet2/alarm-contact',
    label: 'hal-toiletdeur',
    type: 'gate',
    display: { left: 400, top: 350 },
},
{
    topic: 'homie/homey/hal-sensor-magnet2/alarm-contact',
    label: 'hal-toiletdeur',
    type: 'label',
    display: { left: 350, top: 350, prefix: "alarm: " },
},

Three the same topics. Only the type: label is working. Switching nicely between FALSE and TRUE. But the images are not changing. I expect that the image must be closed-door-tiny.png per default. And after an alarm should change to open-door-tiny.png, this is not happening. 

The type gate did not show up. The div id is not generated or shown? There's is an mqtt_1 (door) and a mqtt_3 (label) but no mqtt_2 (gate).  I use the inspect function within Chrome to check and also check the values with a MQTT client. 

Any help is appreciated. 

Regards,
Bert

**Comment by DMailMan** [comment]
Hi Bert
The reason the gate item did not show up is because the type of object is 'gates' and you have called it 'gate' in your config by mistake.
One enhancement could be to check that at least one type has been recognised, and to report a log message when an unknown type is used - but there's no such safety net in place just yet I'm afraid.

But when I fix and use your config, and post an 'open' message to homie/homey/hal-sensor-magnet2/alarm-contact then I get a door and gate image appearing and changing state as expected between open and closed.

![image](https://user-images.githubusercontent.com/3164849/66716037-d8022100-edc1-11e9-9cd9-b834cbc54722.png)

**Comment by bdeelman** [comment]
Hi David,

I need some assistance. 

Yes I made a typo, but you are using door, light (singular) and gates (plural). anyway. ;)

The module is not working like I thought it should be. Only the label part is working for me. That one is switching between true and false or is showing the temperature for instance. 

Every other type (motion, gates) is not working. a light is not showing up. Is showed but with display: none. Just to figure what is happening I made two subscriptions with the same topic. A **door** and a **label**. When the door (I use a Xaomi Window Sensor) is closed the (mqtt) topic value is false. When I open the door the status is true. Sounds resonable it is an alarm. So the alarm is true, the door is open. I did some debugging at the updateDivForItem function. 

I use the console.log functionality. With the door closed I received these values:
closed = false
state = FALSE
image = open-door-tiny.png

This is the other way around as I was expected. closed should be true (door = closed) and image should be closed-door-tiny.png. I played around with the ternary operator, I switched Open and Closed. But the image is still the same.  

The label part is working. Nicely switch between false and true. 

Can you help me with this one? 

Regards,
Bert

**Comment by bertdeelman** [comment]
Hi,

Beside of my typo, it is not working like I think it should work. I did a little rewrite and made it work (for me). I used the Material Design Icons (http://materialdesignicons.com/) and did not use images anymore. Also replaced the separate door, gates and light configuration and used only two: device and label. 

Thanks for your inspiration! 

![Schermafbeelding 2020-04-22 om 19 28 19](https://user-images.githubusercontent.com/44754190/80013622-7b4d9480-84cf-11ea-8181-58d320535a25.png)
